bool catchCommand
float32[] ee